iceoryx_posh/source/popo/ports/client_port_roudi.cpp

```diff
@@ -38,23 +40,144 @@ ClientPortRouDi::MemberType_t* ClientPortRouDi::getMembers() noexcept
     return reinterpret_cast<MemberType_t*>(BasePort::getMembers());
 }
 
+QueueFullPolicy2 ClientPortRouDi::getResponseQueueFullPolicy() const noexcept
+{
+    return static_cast<QueueFullPolicy2>(getMembers()->m_chunkReceiverData.m_queueFullPolicy);
+}
+
 cxx::optional<capro::CaproMessage> ClientPortRouDi::tryGetCaProMessage() noexcept
 {
-    /// @todo
+    // get subscribe request from user side
+    const auto currentConnectRequest = getMembers()->m_connectRequested.load(std::memory_order_relaxed);
 
-    // nothing to change
-    return cxx::nullopt_t();
+    const auto currentConnectionState = getMembers()->m_connectionState.load(std::memory_order_relaxed);
+
+    if (currentConnectRequest && (ConnectionState::NOT_CONNECTED == currentConnectionState))
+    {
+        getMembers()->m_connectionState.store(ConnectionState::CONNECT_REQUESTED, std::memory_order_relaxed);
+
+        capro::CaproMessage caproMessage(capro::CaproMessageType::CONNECT,
+                                         BasePort::getMembers()->m_serviceDescription);
+
+        return cxx::make_optional<capro::CaproMessage>(caproMessage);
+    }
+    else if (!currentConnectRequest && (ConnectionState::CONNECT_REQUESTED == currentConnectionState))
+    {
+        getMembers()->m_connectionState.store(ConnectionState::NOT_CONNECTED, std::memory_order_relaxed);
+
+        // nothing to change
+        return cxx::nullopt_t();
+    }
+    else if (!currentConnectRequest && (ConnectionState::CONNECTED == currentConnectionState))
+    {
+        getMembers()->m_connectionState.store(ConnectionState::DISCONNECT_REQUESTED, std::memory_order_relaxed);
+
+        capro::CaproMessage caproMessage(capro::CaproMessageType::DISCONNECT,
+                                         BasePort::getMembers()->m_serviceDescription);
+        caproMessage.m_chunkQueueData = static_cast<void*>(&getMembers()->m_chunkReceiverData);
```

Comment:
Yeah, there can be multiple queue types and we cannot use a virtual base class because of the SHM. But it would be possible to use an empty base class and pass a pointer to this class around. It would be a bit more type safe since we at least know that the pointer belongs to a queue and cannot be anything.